src/pages/distroSettings/tabs/HostTab/getFormSchema.tsx

```diff
@@ -1,24 +1,58 @@
-import { InlineCode } from "@leafygreen-ui/typography";
 import { GetFormSchema } from "components/SpruceForm";
 import { CardFieldTemplate } from "components/SpruceForm/FieldTemplates";
+import { Arch, BootstrapMethod, Provider, SshKey } from "gql/generated/types";
 import {
-  Arch,
-  BootstrapMethod,
-  CommunicationMethod,
-  FeedbackRule,
-  HostAllocatorVersion,
-  OverallocatedRule,
-  Provider,
-  RoundingRule,
-  SshKey,
-} from "gql/generated/types";
+  linuxArchitectures,
+  nonWindowsArchitectures,
+  windowsArchitectures,
+} from "./constants";
+import {
+  acceptableHostIdleTime,
+  arch,
+  authorizedKeysFile,
+  bootstrapMethod,
+  clientDir,
+  communicationMethod,
+  env,
+  feedbackRule,
+  futureHostFraction,
+  homeVolumeFormatCommand,
+  hostsOverallocatedRule,
+  icecreamConfigPath,
+  icecreamSchedulerHost,
+  isVirtualWorkStation,
+  jasperBinaryDir,
+  jasperCredentialsPath,
+  lockedMemoryKb,
+  maximumHosts,
+  minimumHosts,
+  numFiles,
+  numProcesses,
+  numTasks,
+  preconditionScripts,
+  rootDir,
+  roundingRule,
+  serviceUser,
+  setupAsSudo,
+  setupScript,
+  shellPath,
+  sshKey,
+  sshOptions,
+  user,
+  userSpawnAllowed,
+  version,
+  virtualMemoryKb,
+  workDir,
+} from "./schemaFields";
 
 type FormSchemaParams = {
+  architecture: Arch;
   provider: Provider;
   sshKeys: SshKey[];
 };
 
```

Comment:
I'm not sure what can be done about this, but I notice I can save with an invalid configuration:

https://github.com/evergreen-ci/spruce/assets/47064971/b0c57f7a-e154-49d5-8656-ab4104e21480

If it's too difficult to address on the frontend, possibly we can just add stronger validation on the backend 😅

Reply:
Good catch, would like to have airtight validation on the back end so I'll add that in EVG-20943!